### PR TITLE
oap-kernel: Fix for service that implements collection interface

### PR DESCRIPTION
**Pull request by nofateg**

### CODE REVIEW

**Comment by vdiravka** [comment]
* Fix checkstyle error.
* Add description to the PR

File: oap-stdlib/src/main/java/oap/application/Kernel.java [hunk 1]
```diff
@@ -295,7 +301,8 @@ private void linkLinks( ServiceInitialization initialization ) {
 
                         var linkMethod = reflect.method( "add" + methodSuffix ).orElse( null );
                         if( linkMethod == null ) linkMethod = reflect.method( "set" + methodSuffix ).orElse( null );
-                        if( linkMethod == null ) linkMethod = reflect.method( "add" + methodSuffix + "Listener" ).orElse( null );
+                        if( linkMethod == null )
```
Comment (vdiravka) [inline oap-stdlib/src/main/java/oap/application/Kernel.java:304]: it is easy to make the mistake here in if-else statements now or later in the process of refactoring. 
Please add curly braces, they also will help to understand the code better.

File: oap-stdlib/src/main/java/oap/application/KernelHelper.java [hunk 2]
```diff
@@ -171,4 +176,12 @@ public static void restoreThreadName() {
         }
 
     }
+
+    record Parameter(Object value, boolean ignoreCast) {
+    }
+
+    static class Parameters {
```
Comment (vdiravka) [inline oap-stdlib/src/main/java/oap/application/KernelHelper.java:183]: Where we can find the list of all supported `configurationParameters` and `serviceParameters`? Or they are declared in config files itself?

File: oap-stdlib/src/main/java/oap/application/KernelHelper.java [hunk 3]
```diff
@@ -171,4 +176,12 @@ public static void restoreThreadName() {
...
+        public final LinkedHashMap<String, Object> configurationParameters = new LinkedHashMap<>();
+        public final LinkedHashMap<String, Object> serviceParameters = new LinkedHashMap<>();
```
Comment (vdiravka) [inline oap-stdlib/src/main/java/oap/application/KernelHelper.java:185]: What is the difference between `configurationParameters` and `serviceParameters`?

File: oap-stdlib/src/main/java/oap/application/Kernel.java [hunk 4]
```diff
@@ -306,15 +313,16 @@ private void linkLinks( ServiceInitialization initialization ) {
...
-                            else exception( new ReflectException( "link to " + service.implementationName + "/" + service.service.implementation
-                                + " should have field " + fieldName
-                                + " for " + initialization.implementationName + "/" + initialization.service.implementation ) );
+                            else
+                                exception( new ReflectException( "link to " + service.implementationName + "/" + service.service.implementation
+                                    + " should have field " + fieldName
+                                    + " for " + initialization.implementationName + "/" + initialization.service.implementation ) );
                         }
                     },
                     exception( e -> new ApplicationException( "Unknown service link " + serviceRef + " in"
```
[the remark this answers is not in the document]
Comment (nofateg) [reply oap-stdlib/src/main/java/oap/application/Kernel.java:328]: concatenation is easier to read

File: oap-stdlib/src/main/java/oap/application/KernelHelper.java (hunk 2 above)
Comment (nofateg) [reply oap-stdlib/src/main/java/oap/application/KernelHelper.java:183]: Parameters -> ServiceConfigurationParameters
Parameter -> ServiceConfigurationParameter

**Pull request merged**